CollAction/Controllers/ProjectsController.cs

```diff
@@ -274,14 +276,33 @@ public async Task<IActionResult> Commit(int id, CommitProjectViewModel commitPro
                     "</span>";
                 string subject = $"Thank you for participating in the \"{commitProjectViewModel.ProjectName}\" project on CollAction";
                 _emailSender.SendEmail(user.Email, subject, confirmationEmail);
-                return View("ThankYouCommit", commitProjectViewModel);
+                return LocalRedirect($"~/Projects/{Uri.EscapeDataString(commitProjectViewModel.ProjectName.NormalizeUriPart())}/{commitProjectViewModel.ProjectId}/thankyou");
             }
             else
             {
                 return View("Error");
             }
         }
 
+        [Authorize]
+        [HttpGet]
+        public IActionResult ThankYouCommit(int id, string name)
+        {
+            if (name != null)
```

Comment:
I don't think this is really necessary to validate data we're not going to use anyway (see below).

Reply:
Fixed